core/sort/sort.service.js

```diff
@@ -22,4 +22,58 @@ export function orderFactory(model) {
 
 		return by;
 	};
+}
+
+function newKey(pair) {
+	let key = '';
+	if (!isString(pair)) {
+		key = Object.keys(pair)[0];
+	} else {
+		key = pair.split(/[+-]/)[1];
+	}
+	
+	if(!key){
+		throw new AppError(
+			'pair',
+			`Key is not defined in "${pair}"`);
+	}
+
+	return key;
+}
+
+function newValue(pair) {
+	let value = '';
```

Comment:
consider to use 2 returns instead of `value`, in this case it will separate logic better